src/NHibernate/Engine/ForeignKeys.cs

```diff
@@ -246,16 +245,6 @@ public static object GetEntityIdentifierIfNotUnsaved(string entityName, object e
 
 					if (IsTransientFast(entityName, entity, session).GetValueOrDefault())
 					{
-						/***********************************************/
-						// TODO NH verify the behavior of NH607 test
-						// these lines are only to pass test NH607 during PersistenceContext porting
-						// i'm not secure that NH607 is a test for a right behavior
-						EntityEntry entry = session.PersistenceContext.GetEntry(entity);
-						if (entry != null)
-							return entry.Id;
```

Comment:
PersistenceContext can contain transient objects in state Saving. This hack was added (that returns null ID for transient object) to make `GetOrphans` work. Removing it as `GetOrphans` is not relying on `GetEntityIdentifierIfNotUnsaved` anymore.